middle_end/inline_and_simplify.ml

```diff
@@ -1293,10 +1295,12 @@ and simplify env r (tree : Flambda.t) : Flambda.t * R.t =
         lam, R.map_benefit r B.remove_branch
       | _ ->
         let env = E.inside_branch env in
+        let really_import_approx = E.really_import_approx env in
         let f (i, v) (acc, r) =
           let approx = R.approx r in
           let lam, r = simplify env r v in
-          ((i, lam)::acc, R.set_approx r (A.meet (R.approx r) approx))
+          ((i, lam)::acc,
+           R.set_approx r (A.meet ~really_import_approx (R.approx r) approx))
```

Comment:
Needs another space of indentation (and below, please indent by 2 spaces when "Some l, R.set_approx ..." is split across two lines).